src/api/schoolsMapInfo.ts

```diff
@@ -0,0 +1,72 @@
+import { gql, LazyQueryResult, useLazyQuery } from '@apollo/client';
+import { convertFilterStateToGraphQLVariables, FiltersState } from '../utils/filters';
+import { ISchoolListingQuery, ISchoolListingQueryVariables } from '../types/graphql';
+import { SCHOOL_CARD_PROPS_FRAGMENT } from '../components/SchoolCard';
+import { getPaginationOffset, PER_PAGE } from '../utils/pagination';
+import { FilterKey } from '../data/filters';
+
+const SCHOOLS_LISTING_QUERY = gql`
+  ${SCHOOL_CARD_PROPS_FRAGMENT}
+  query SchoolMapInfo(
+    $offset: Int
+    $first: Int
+    $query: String
+    $isPublic: Boolean
+    $extendedSubjects: [String]
+    $districts: [String]
+  ) {
+    allSchools(
+      offset: $offset
+      first: $first
+      query: $query
+      isPublic: $isPublic
+      extendedSubjects: $extendedSubjects
+      districts: $districts
+    ) {
+      totalCount
+      edges {
+        node {
+          ...SchoolCardProps
+        }
+      }
+    }
+  }
+`;
+
+type GetSchoolListingFn = (
```

Comment:
I suggest to change the name of type, so we could easily know it is a type, e.g.: TGetSchoolListing or GetSchoolListingType